src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -118,47 +118,62 @@ internal sealed class BuildServerConnection
             var clientDir = buildPaths.ClientDirectory;
             var timeoutNewProcess = timeoutOverride ?? TimeOutMsNewProcess;
             var timeoutExistingProcess = timeoutOverride ?? TimeOutMsExistingProcess;
-            var clientMutexName = GetClientMutexName(pipeName);
             Task<NamedPipeClientStream> pipeTask = null;
-            using (var clientMutex = new Mutex(initiallyOwned: true,
```

Comment:
Why not keep the using statement and wrap it in a Try/Finally?

Reply:
Assume you meant "try / catch" there. 

I wanted to be very specific about what error I was catching here: the Mutex throwing. Only way to do that was to remove the `using` and use a `try / catch` in this manner.